pkg/internal/vault/vault.go

```diff
@@ -164,7 +168,17 @@ func (v *Vault) setToken(client Client) error {
 		return nil
 	}
 
-	return fmt.Errorf("error initializing Vault client tokenSecretRef or appRoleSecretRef not set")
+	kubernetesAuth := v.issuer.GetSpec().Vault.Auth.Kubernetes
+	if kubernetesAuth.Role != "" {
```

Comment:
What guarantees that `kubernetesAuth` is non-nil?

Reply:
Also, this field is required - if `Kubernetes` is non-nil, this field should *always* be set, no?

Reply:
None of the auth types are pointers
```
	TokenSecretRef SecretKeySelector `json:"tokenSecretRef,omitempty"`

	// This Secret contains a AppRole and Secret
	// +optional
	AppRole VaultAppRole `json:"appRole,omitempty"`

	// This contains a Role and Secret with a ServiceAccount token to
	// authenticate with vault.
	// +optional
	Kubernetes KubernetesAuth `json:"kubernetes,omitempty"`
}
```